### PR TITLE
Add platform_version to test Obs integration data

### PR DESCRIPTION
**Pull request by 07souravkunda**
## Proposed changes
Adding platform_version to BrowserStack test observability integration data

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by kamal-kaur04 — APPROVED** [review]

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 1]
```diff
@@ -866,10 +870,26 @@ class _InsightsHandler {
...
+    private getPlatformVersion() {
+        const caps = (this._userCaps as WebdriverIO.Capabilities)
+        const bstackOptions = (this._userCaps as WebdriverIO.Capabilities)?.['bstack:options']
+        const keys = ['platformVersion', 'platform_version', 'osVersion', 'os_version']
+
+        for (const key of keys) {
+            if ( bstackOptions && bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]) {
+                return bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]
+            } else if ((caps as WebdriverIO.Capabilities)[key as keyof WebdriverIO.Capabilities]) {
+                return (caps as WebdriverIO.Capabilities)[key as keyof WebdriverIO.Capabilities]
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/insights-handler.ts:887]: these have been already casted above
```suggestion
                return caps[key as keyof WebdriverIO.Capabilities]
```

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 2]
```diff
@@ -866,10 +870,26 @@ class _InsightsHandler {
...
 
+    private getPlatformVersion() {
+        const caps = (this._userCaps as WebdriverIO.Capabilities)
+        const bstackOptions = (this._userCaps as WebdriverIO.Capabilities)?.['bstack:options']
+        const keys = ['platformVersion', 'platform_version', 'osVersion', 'os_version']
+
+        for (const key of keys) {
+            if ( bstackOptions && bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]) {
+                return bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]
+            } else if ((caps as WebdriverIO.Capabilities)[key as keyof WebdriverIO.Capabilities]) {
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/insights-handler.ts:886]: same
```suggestion
            } else if (caps[key as keyof WebdriverIO.Capabilities]) {
```

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 3]
```diff
@@ -866,10 +870,26 @@ class _InsightsHandler {
...
         }
     }
 
+    private getPlatformVersion() {
+        const caps = (this._userCaps as WebdriverIO.Capabilities)
+        const bstackOptions = (this._userCaps as WebdriverIO.Capabilities)?.['bstack:options']
+        const keys = ['platformVersion', 'platform_version', 'osVersion', 'os_version']
+
+        for (const key of keys) {
+            if ( bstackOptions && bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]) {
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/insights-handler.ts:884]:
```suggestion
            if (bstackOptions && bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]) {
```

Comment (07souravkunda) [reply packages/wdio-browserstack-service/src/insights-handler.ts]: done

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**